arangod/Cluster/Maintenance.cpp

```diff
@@ -847,7 +851,16 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
           {
             VPackObjectBuilder o(&report);
             report.add(OP, VP_SET);
+            // Report new current entry ...
             report.add("payload", localCollectionInfo.slice());
+            // ... if and only if plan for this shard has changed in the meantime
+            report.add(VPackValue("precondition"));
+            {
+              VPackObjectBuilder p(&report);
+              report.add(
+                PLAN_COLLECTIONS + dbName + "/" + colName + "/shards/" + shName,
+                pdbs.get(std::vector<std::string>{dbName, colName, "shards", shName}));
```

Comment:
This is a pure Vpack get() call.  It could throw.  Need try/catch or one of the alternate get methods that return a neutral default.

Reply:
Good catch, will amend. I have a feeling that this was already there before I started, but it of course needs to be fixed.

Reply:
This was actually added in this PR, contrary to my memory. And I have fixed it by wrapping it into a try catch. The case should be very rare and only occur if the collection is dropped in the plan whilst reporting to Current.